src-utils/HydraAuctionUtils/Tx/AutoCreateTx.hs

```diff
@@ -91,7 +91,7 @@ data AutoCreateParams = AutoCreateParams
   , outs :: [TxOut CtxTx]
   , toMint :: TxMintValue BuildTx
   , changeAddress :: Address ShelleyAddr
-  , validityBound :: (Maybe POSIXTime, Maybe POSIXTime)
+  , validityBound :: Interval POSIXTime
```

Comment:
Interesting solution.

On one side we reuse polymorphic code, on the other side Interval reuse intuitively feels problems prone to me.

Reply:
Like quote from PR comment:

> and for v3 we scrap this Interval coming up with a better (non-backwards compatible) Interval definition.

Overall their backwards comparability policy does not inspire confidence, so I would not depend on Plutus more then necessary. But I guess it that case it should be fine.